source/common/io/io_uring_worker.h

```diff
@@ -0,0 +1,486 @@
+#pragma once
+
+#include "io_uring.h"
+#include "source/common/buffer/buffer_impl.h"
+#include "source/common/common/logger.h"
+
+#include "source/common/io/io_uring.h"
+#include "source/common/io/io_uring_impl.h"
+
+namespace Envoy {
+namespace Io {
+
+class IoUringAcceptSocket : public IoUringSocket, protected Logger::Loggable<Logger::Id::io> {
+public:
+  IoUringAcceptSocket(os_fd_t fd, IoUringHandler& io_uring_handler, IoUringWorker& parent) :
+    fd_(fd), io_uring_handler_(io_uring_handler), parent_(parent) {}
+
+  // IoUringSocket
+  os_fd_t fd() const override { return fd_; }
+  void start() override;
+  void close() override;
+  void enable() override;
+  void disable() override;
+
+  void onAccept(int32_t result) override;
+  void onClose(int32_t result) override;
+  void onCancel(int32_t result) override;
+private:
+  os_fd_t fd_;
+
+  IoUringHandler& io_uring_handler_;
+  IoUringWorker& parent_;
+
+  sockaddr_storage remote_addr_;
+  socklen_t remote_addr_len_{sizeof(remote_addr_)};
+  os_fd_t connection_fd_{INVALID_SOCKET};
```

Comment:
What's the difference between fd_ and connection_fd_?

Reply:
`fd_` is the socket we listening on. The `connection_fd_` store the socket which returns by accepted. https://github.com/envoyproxy/envoy/pull/24305/files/2013f76b1f070b655e2e7e90f60d74c760e6c0f3#diff-5a267b3ed3d6b9001c2748d9940424e4582512c97271f77664dd0e126af81160R27

Then IoUringSocketHandle will return `connection_fd_` when the upper layer code invokes the IoUringSocketHandleImpl::accept